amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java

```diff
@@ -86,7 +89,20 @@ public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionExcep
   @Override
   public boolean dropNamespace(String[] namespace, boolean cascade)
       throws NoSuchNamespaceException, NonEmptyNamespaceException {
-    return false;
+    String database = namespaceToDatabase(namespace);
+    if (!unifiedCatalog.databaseExists(database)) {
+      throw new NoSuchNamespaceException(namespace);
+    }
+    List<TableIDWithFormat> tables = unifiedCatalog.listTables(database);
+    if (!tables.isEmpty() && !cascade) {
```

Comment:
Why not drop all tables first and then drop the database?

I think should this logic be implemented directly by the parent (spark-common).
We already have concrete logic in  `SparkUnifiedCatalogBase.dropNamespace`.
so，remove Override is enough.

WDYT?

Reply:
I agreed that we may reuse some implementation in the parent class. But I think it is dangerous to delete all tables by default in Spark 3.2 implementation. So wee may:

- Add a method in the base class to support dropping namespace cascade or not.
- Drop namespace by default with false for Spark 3.2